src/azure-account.ts

```diff
@@ -773,17 +785,47 @@ async function getEnvironments(): Promise<Environment[]> {
 
 	const config = workspace.getConfiguration('azure');
 	const ppe = config.get<Environment>('ppe');
-	if (ppe) {
-		return [
-			...staticEnvironments,
-			{
-				...ppe,
-				name: azurePPE
-			}
-		]
-	} else {
-		return staticEnvironments;
+	const apiProfile = config.get<Boolean>('target_azurestack_api_profile');
+	try {
+		if (ppe) {
+			const activeDirectoryUrl = ppe.activeDirectoryEndpointUrl.endsWith('/') ? ppe.activeDirectoryEndpointUrl.slice(0,-1) : ppe.activeDirectoryEndpointUrl;
+			const validateAuthority = activeDirectoryUrl.endsWith('/adfs') ? false : true;
+			if (apiProfile) {
+				let resourceMangerUrl = ppe.resourceManagerEndpointUrl;
+				resourceMangerUrl = resourceMangerUrl.endsWith('/') ? resourceMangerUrl.slice(0,-1) : resourceMangerUrl;
+				const endpointSuffix = '/metadata/endpoints';
+				const apiVersion = '2018-05-01';
+				const ppeResponse = await fetch(`${resourceMangerUrl}${endpointSuffix}?api-version=${apiVersion}`);
+				if (ppeResponse.ok) {
+					const ppeMetadata: IPPEMetaData = await ppeResponse.json();
+					return [
+						...staticEnvironments,
+						{
+							...ppe,
+							name: azurePPE,
+							portalUrl: ppeMetadata.portalEndpoint,
+							galleryEndpointUrl: ppeMetadata.galleryEndpoint,
+							activeDirectoryGraphResourceId: ppeMetadata.graphEndpoint,
+							storageEndpointSuffix: resourceMangerUrl.substring(resourceMangerUrl.indexOf('.')),
+							keyVaultDnsSuffix: '.vault'.concat(resourceMangerUrl.substring(resourceMangerUrl.indexOf('.'))),
+							managementEndpointUrl: ppeMetadata.authentication.audiences[0],
+							validateAuthority: validateAuthority
+						}]
+				}
+				return [
```

Comment:
We also need to return this if `apiProfile` is false

Reply:
@RMacfarlane what do we want to return if apiProfile is false?

Reply:
I feel like the if(apiProfile) check is not needed at all here, as the response we are returning should be applicable for all other custom PPE as well, just that we will be using the older api version to get the metadata

Reply:
Azure Stack is not the only thing that can change the PPE setting, if others are using it to test something, we need to preserver the original behavior and return the list of environments as before